File: code/backend/src/main/java/br/com/anagropets/dto/registrovenda/RegistroVendaCadastroDTO.java
package br.com.anagropets.dto.registrovenda;

import java.math.BigDecimal;
import java.util.List;

import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class RegistroVendaCadastroDTO {
	private @NotNull Long idCliente;
	private @NotNull BigDecimal valorTotal;
	private @NotNull List<ItemVendaCadastroDTO> itemVendaList;
	private @NotNull Long idMetodoPagamento;
	private @NotNull Integer parcelasPagamento;
}

File: code/frontend/src/app/models/dto/RegistroVendaCadastroDTO.ts
import { ItemVendaCadastroDTO } from './ItemVendaCadastroDTO';

export interface RegistroVendaCadastroDTO {
  idCliente: number;
  valorTotal: number;
  itemVendaList: Array<ItemVendaCadastroDTO>;
  idMetodoPagamento: number;
  parcelasPagamento: number;
}
